zarr/meta.py

```diff
@@ -92,7 +92,7 @@ class Metadata2:
     ZARR_FORMAT = ZARR_FORMAT
 
     @classmethod
-    def parse_metadata(cls, s: Union[MappingType, str]) -> MappingType[str, Any]:
+    def parse_metadata(cls, s: Union[MappingType, bytes]) -> MappingType[str, Any]:
```

Comment:
Should this still include `str` as an option? Same question with the functions below

```suggestion
    def parse_metadata(cls, s: Union[MappingType, bytes, str]) -> MappingType[str, Any]:
```